### PR TITLE
feat(issue-stream): Modify issue stream title to only use two rows

### PR DESCRIPTION
**Pull request by malwilley**
Ref https://github.com/getsentry/sentry/issues/77822

Previous layout:

```
**<title>** <location>
<error_type> | <error_message>
<project> <seen> <other things>
```

New layout:

```
**<title>** <error_message>
<project> | <error_type> | <location> | <other things>
```

Note that this does most of the work, but there are a couple things that still need polishing:

- Extra details line has some extra spaces due to the replay counts component returning null sometimes
- Extra details should all be gray text

Before:

CleanShot 2024-10-10 at 17 08 33@2x

After:

CleanShot 2024-10-10 at 17 08 51@2x

### CODE REVIEW

**Comment by JoshFerge** [comment]
we changing the colors?

**Comment by malwilley** [comment]
> we changing the colors?

Yes - you can [see the designs here](https://www.figma.com/design/MM42MEY0OiZRVzQlI8XHej/Q1-Issue-Stream?node-id=2261-39331&node-type=frame&t=XY8je5s7gM3XXJop-0)

Though it's only under a feature flag right now and things will probably change a bit before it gets released

**Comment by JoshFerge** [comment]
got it -- i think we should try and have more contrast between the issue titles and everything else on the page, and the new designs reduce that a bit.

**Review by scttcper — APPROVED** [review]
🚫 color

File: static/app/components/eventOrGroupExtraDetails.tsx [hunk 1]
```diff
@@ -74,66 +78,86 @@ function EventOrGroupExtraDetails({
...
+  const items = [
+    shortId ? (
+      <InboxShortId
+        shortId={shortId}
+        avatar={
+          project && <ShadowlessProjectBadge project={project} avatarSize={12} hideName />
+        }
+      />
+    ) : null,
+    hasNewLayout && level ? <ErrorLevelText level={level} /> : null,
```
Comment (MichaelSun48) [inline static/app/components/eventOrGroupExtraDetails.tsx:95]: Nit: the color of this text semes to be solid black, but the designs appear to be `subtext` colored. 

Current: 
image

Designs:
image

File: static/app/components/eventOrGroupHeader.tsx [hunk 2]
```diff
@@ -140,6 +145,7 @@ const truncateStyles = css`
 
 const Title = styled('div')`
```
Comment (MichaelSun48) [inline static/app/components/eventOrGroupHeader.tsx:146]: Nit: might need a bit more margin right to align it with the header divider
Screenshot 2024-10-10 at 3 39 14 PM

**Review by MichaelSun48 — APPROVED** [review]
Couple minor nits

File: static/app/components/eventOrGroupExtraDetails.tsx (hunk 1 above)
Comment (malwilley) [reply static/app/components/eventOrGroupExtraDetails.tsx:95]: Good catch! Yeah there are a few things in that row that aren't colored correctly, but the PR was getting pretty large so I decided to leave that work for later

**Pull request merged**